tests/v2/validation/snapshot/snapshot.go

```diff
@@ -115,49 +120,49 @@ func snapshotRestore(t *testing.T, client *rancher.Client, clusterName string, e
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(ServiceType).Delete(serviceResp)
+	err = steveclient.SteveType(serviceType).Delete(serviceResp)
 	require.NoError(t, err)
 
-	err = steveclient.SteveType(IngressSteveType).Delete(ingressResp)
+	err = steveclient.SteveType(ingressSteveType).Delete(ingressResp)
 	require.NoError(t, err)
 }
 
-func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) initialSnapshotConfig {
+func snapshotRKE1(t *testing.T, client *rancher.Client, podTemplate corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID, localClusterID string, etcdRestore *etcdsnapshot.Config, isRKE1 bool) (*management.Cluster, initialSnapshotConfig) {
+	if etcdRestore.ReplaceNodes != (etcdsnapshot.ReplaceNodes{}) {
+		scaling.ReplaceRKE1Nodes(t, client, clusterName, etcdRestore.Etcd, etcdRestore.ControlPlane, etcdRestore.Worker)
+	}
+
 	existingSnapshots, err := etcdsnapshot.GetRKE1Snapshots(client, clusterID)
 	require.NoError(t, err)
 
 	err = etcdsnapshot.CreateRKE1Snapshot(client, clusterName)
 	require.NoError(t, err)
 
-	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	restoreSnapshotResp, err := client.Management.Cluster.ByID(clusterID)
 	require.NoError(t, err)
 
 	podErrors := pods.StatusPods(client, clusterID)
 	assert.Empty(t, podErrors)
 
-	if etcdRestore.ReplaceWorkerNode {
-		scaling.ReplaceRKE1Nodes(t, client, clusterName, false, false, true)
-	}
-
-	initialKubernetesVersion := clusterResp.RancherKubernetesEngineConfig.Version
-	require.Equal(t, initialKubernetesVersion, clusterResp.RancherKubernetesEngineConfig.Version)
+	initialKubernetesVersion := restoreSnapshotResp.RancherKubernetesEngineConfig.Version
+	require.Equal(t, initialKubernetesVersion, restoreSnapshotResp.RancherKubernetesEngineConfig.Version)
 
-	initialControlPlaneUnavailable := clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane
-	require.Equal(t, initialControlPlaneUnavailable, clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane)
+	initialControlPlaneUnavailable := restoreSnapshotResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane
+	require.Equal(t, initialControlPlaneUnavailable, restoreSnapshotResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane)
 
-	initialWorkerUnavailableValue := clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker
-	require.Equal(t, initialWorkerUnavailableValue, clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker)
+	initialWorkerUnavailableValue := restoreSnapshotResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker
+	require.Equal(t, initialWorkerUnavailableValue, restoreSnapshotResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker)
```

Comment:
this require is redundant, right? since you've just set it in the var above, it seems unnecessary

Reply:
Resolved